File: src/main/java/ast/Node.java
package ast;

import tree.TargetTreeNode;

import java.util.ArrayList;
import java.util.List;

public class Node implements ASTNode {

	String name;



	public Node(String name) {
		super();
		this.name = name;
	}


	// change list for something less complex
	@Override
	public Object execute(TargetTreeNode targetTreeNode) {
		System.out.println(name);
		List<TargetTreeNode> answer = new ArrayList<>();
		// TODO - remove true
		if( targetTreeNode.getName().equals(name) || true )
		{
			answer.add(targetTreeNode);
		}
		return answer;
	}

}

File: src/main/java/ast/Node.kt
package ast

import java.util.ArrayList
import tree.TargetTreeNode

class Node(var name: String) : ASTNode {
    // change list for something less complex
    override fun execute(targetTreeNode: TargetTreeNode): Any? {
        println(name)
        val answer: MutableList<TargetTreeNode> = ArrayList()
        // TODO - remove true
        if (targetTreeNode.name == name || true) {
            answer.add(targetTreeNode)
        }
        return answer
    }
}
